axon-mongo-spring/src/main/java/org/axonframework/extensions/mongo/spring/SpringMongoTemplate.java

```diff
@@ -0,0 +1,237 @@
+     * Instantiate a Builder to be able to create a {@link SpringMongoTemplate}.
+     * <p>
+     * The {@code domainEventsCollectionName}, {@code snapshotEventsCollectionName},
+     * {@code trackingTokensCollectionName} {@code sagasCollectionName} and {@code deadLetterCollectionName} are
+     * respectively defaulted to {@code trackingtokens}, {@code domainevents}, {@code snapshotevents}, {@code sagas} and
+     * {@code deadletters}.
+     * <p>
+     * The {@link MongoDatabaseFactory} is a <b>hard requirement</b> and as such should be provided.
+     *
+     * @return a Builder to be able to create a {@link DefaultMongoTemplate}
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public MongoCollection<Document> trackingTokensCollection() {
+        return databaseSupplier.get().getCollection(trackingTokensCollectionName);
+    }
+
+    @Override
+    public MongoCollection<Document> eventCollection() {
+        return databaseSupplier.get().getCollection(domainEventsCollectionName);
+    }
+
+    @Override
+    public MongoCollection<Document> snapshotCollection() {
+        return databaseSupplier.get().getCollection(snapshotEventsCollectionName);
+    }
+
+    @Override
+    public MongoCollection<Document> sagaCollection() {
+        return databaseSupplier.get().getCollection(sagasCollectionName);
+    }
+
+    @Override
+    public MongoCollection<Document> deadLetterCollection() {
+        return databaseSupplier.get().getCollection(deadLetterCollectionName);
+    }
+
+    public static class Builder {
+
+        private MongoDatabaseFactory factory;
+        private String databaseName = null;
+        private String trackingTokensCollectionName = "trackingtokens";
+        private String domainEventsCollectionName = "domainevents";
+        private String snapshotEventsCollectionName = "snapshotevents";
+
+        private String sagasCollectionName = "sagas";
+        private String deadLetterCollectionName = "deadletters";
+
+        /**
+         * Sets the {@code factory} to use. Needs to be set in order to create an {@link SpringMongoTemplate}.
+         *
+         * @param factory a {@link MongoDatabaseFactory} used to create a Mongo database
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder factory(MongoDatabaseFactory factory) {
+            assertNonNull(factory, "The factory may not be null");
+            this.factory = factory;
+            return this;
+        }
+
+        /**
+         * Sets the {@code databaseName} to use as the database. Defaults to the default of the {@code factory}.
+         *
+         * @param databaseName a {@link String} specifying the database name
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder databaseName(String databaseName) {
+            assertNonNull(databaseName, "The databaseName may not be null");
```

Comment:
Might be better to use `BuilderUtils#assertNonEmpty`. That'll validate a non-empty and non-null `String` is used.
This pointer counts for all the builder methods in this class ingesting a `String`, by the way.

Reply:
Changed